### PR TITLE
improve refresh account data

### PR DESCRIPTION
**Pull request by marcinbodnar**

### CODE REVIEW

File: src/actions/account.js [hunk 1]
```diff
@@ -3,26 +3,17 @@ import { createActions } from 'redux-actions'
...
 
 export function handleRefreshAccount(history, loader = true) {
    return (dispatch, getState) => {
-      if (loader) {
-         dispatch({
-            type: LOADER_ACCOUNT,
-            loader: true
-         })   
-      }
+      dispatch(loginPending(loader))
```
Comment (vgrichina) [inline src/actions/account.js]: Any reason to not go with async function (similar to wallet actions)? Seems like it'll allow to make this method simpler as request lifecycle management will be handled by middleware and you'll only need to dispatch another action for clearing accounts.

File: src/actions/account.js [hunk 2]
```diff
@@ -42,6 +33,7 @@ export function handleRefreshAccount(history, loader = true) {
                   accounts: wallet.accounts
                }
             })
+            dispatch(loginSuccess(true))
          })
          .catch(e => {
             console.error('Error loading account:', e)
```
Comment (vgrichina) [inline src/actions/account.js]: I think stuff in ` if (e.message && e.message.indexOf('does not exist while viewing') !== -1) {` block should be moved into separate RESET_ACCOUNTS action.

**Comment by marcinbodnar** [comment]
@vgrichina Prepared new version, now we are using readyStatePromise middleware to handle pending status. Also, refreshUrl was moved to redux-actions and I removed some code from wallet.js that will not be used anymore.

File: src/actions/account.js [hunk 3]
```diff
@@ -139,4 +61,29 @@ export const { switchAccount } = createActions({
    SWITCH_ACCOUNT: wallet.selectAccount.bind(wallet)
 })
 
-export const { loginPending, loginSuccess, loginError } = createActions('LOGIN_PENDING', 'LOGIN_SUCCESS', 'LOGIN_ERROR')
+export const { refreshAccount, refreshUrl, resetAccounts } = createActions({
+   REFRESH_ACCOUNT: [
+      wallet.loadAccount.bind(wallet),
+      (loader) => ({ 
+         loader,
```
Comment (vgrichina) [inline src/actions/account.js]: how does this work? Is it generating some `meta` for action? I think we need to standardize the way we pass `meta` to allow generic loader/error handling.

**Comment by marcinbodnar** [comment]
@vgrichina this PR can be closed due to #390 changes

**Pull request closed without merging**